modules/deployment/src/main/java/org/eclipse/xpanse/modules/deployment/ServiceConfigurationUpdateHandler.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ *
+ */
+
+package org.eclipse.xpanse.modules.deployment;
+
+import jakarta.annotation.Resource;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.LockModeType;
+import jakarta.transaction.Transactional;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.xpanse.modules.database.serviceconfiguration.update.ServiceConfigurationUpdateRequest;
+import org.eclipse.xpanse.modules.database.serviceconfiguration.update.ServiceConfigurationUpdateStorage;
+import org.eclipse.xpanse.modules.models.serviceconfiguration.enums.ServiceConfigurationStatus;
+import org.hibernate.exception.LockTimeoutException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+/**
+ * Service configuration more detailed request handler.
+ */
+@Slf4j
+@Component
+public class ServiceConfigurationUpdateHandler {
+
+    public static final String JAVAX_PERSISTENCE_LOCK_TIMEOUT = "javax.persistence.lock.timeout";
+
+    @Value("${javax.persistence.lock.timeout:2000}")
+    private int lockTimeout;
+
+    @Resource
+    private EntityManager entityManager;
+
+    @Resource
+    private ServiceConfigurationUpdateStorage serviceConfigurationUpdateStorage;
+
+    /**
+     * Update service configuration update request.
+     */
+    @Transactional
+    public ServiceConfigurationUpdateRequest
+            updateServiceConfigurationUpdateRequest(UUID id, String resourceName) {
+        try {
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(JAVAX_PERSISTENCE_LOCK_TIMEOUT, lockTimeout);
+            ServiceConfigurationUpdateRequest request = entityManager
+                    .find(ServiceConfigurationUpdateRequest.class, id,
+                            LockModeType.PESSIMISTIC_WRITE, properties);
```

Comment:
this is a wrong place. The lock must be at the read query. The name is a bit misleading. Even though the lockType is PESSIMISTIC_WRITE, it actually blocks both read and write. 
So the first query that is executed must lock the rows for read and write both. 
So any other query which comes after that, must wait until the locks are released. 

You can see some examples here. 
https://beratyesbek.medium.com/pessimistic-and-optimistic-locking-in-jpa-spring-boot-aa59f2b906fa